camunda-bpm-camel-common/src/main/java/org/camunda/bpm/camel/common/CamelServiceCommonImpl.java

```diff
@@ -97,8 +100,28 @@ private Object sendToInternal(String endpointUri,
     }
     exchange.getIn().setBody(variablesToSend);
     exchange.setPattern(ExchangePattern.InOut);
-    Exchange send = producerTemplate.send(endpointUri, exchange);
-    return send.getIn().getBody();
+    Exchange send = producerTemplate.send(endpointUri, exchange);       
+    
+    // Exception handling
+    //    Propogate BpmnError back from camel route, 
+    //    all other exceptions will cause workflow to stop as a technical error
+    // https://docs.camunda.org/get-started/rpa/error-handling/
+    // https://docs.camunda.org/manual/7.15/reference/bpmn20/events/error-events/
+    if (null != send.getException()){
+      // Explicit BPMN business error, workflow has a chance to handle on boundry event, throw as is
+      if (send.getException() instanceof BpmnError) throw ((BpmnError)send.getException());
+      
+      // Unchecked, consider technical error causing BPMN workflow to stop
```

Comment:
Is this really necessary? I think lines 117 and 121 are doing exactly the same thing - as the cast to a runtime exception will not change anything here - r am I missing something?

Reply:
Hi
Ye, agreed, this is pretty much the same.
I had the branch treating the two different exception types as I was undecided if we should only stop the workflow on a Runtime exception and then map Exception to BPMError.

So ye .. we could just keep line 121 and remove
      // Unchecked, consider technical error causing BPMN workflow to stop
      if (send.getException() instanceof RuntimeException) {
        // Technical error (Fails workflow)
        throw (RuntimeException)send.getException();
      }

.. unless wanting to treat RuntimeException differently to Exception.